wgsl/index.bs

```diff
@@ -3247,12 +3247,10 @@ and third clauses and in the body of the `for` statement.
 
 An instance of a function scope variable is a [=dynamic context=].
 Each variable that is [=in scope=] for some invocation has an overlapping
-[=lifetime=] and, therefore, has non-overlapping storage.
-Variables with non-overlapping lifetimes may reuse the storage of previous
+[=lifetime=] and, therefore, has non-overlapping memory.
+Variables with non-overlapping lifetimes may reuse the memory of previous
 variables; however, new instances of the same variable are not guaranteed to
-use the same storage.
-
-## Never-alias assumption TODO ## {#never-alias-assumption}
```

Comment:
My main aim is to delete this TODO.  The section is not needed, because it's already covered.